assets/src/admin/site-scan-notice/notice.js

```diff
@@ -0,0 +1,124 @@
+import { isExternalUrl } from '../../common/helpers/is-external-url';
+import { PluginsWithAmpIncompatibility } from './plugins-with-amp-incompatibility';
+
+// Define Plugin Suppression link.
+const PLUGIN_SUPPRESSION_LINK = new URL( SETTINGS_LINK );
+PLUGIN_SUPPRESSION_LINK.hash = 'plugin-suppression';
+
+export function SiteScanNotice() {
+	const {
+		cancelSiteScan,
+		fetchScannableUrls,
+		isCancelled,
+		isCompleted,
+		isFailed,
+		isInitializing,
+		isReady,
+		pluginsWithAmpIncompatibility,
+		startSiteScan,
+	} = useContext( SiteScan );
+	const { developerToolsOption } = useContext( User );
+	const userIsTechnical = useMemo( () => developerToolsOption === true, [ developerToolsOption ] );
+
+	// Cancel scan on component unmount.
+	useEffect( () => cancelSiteScan, [ cancelSiteScan ] );
+
+	// Fetch scannable URLs on mount. Start site scan right after the component is mounted and the scanner is ready.
+	useEffect( () => {
+		if ( isInitializing ) {
+			fetchScannableUrls();
+		} else if ( isReady ) {
+			startSiteScan();
+		}
+	}, [ fetchScannableUrls, isInitializing, isReady, startSiteScan ] );
+
+	const commonNoticeProps = {
+		className: 'amp-site-scan-notice',
+		isDismissible: true,
+		onDismiss: cancelSiteScan,
+	};
+
+	if ( isFailed || isCancelled ) {
+		return (
+			<AdminNotice type={ ADMIN_NOTICE_TYPE_ERROR } { ...commonNoticeProps }>
+				<p>
+					{ __( 'AMP plugin could not check your site for compatibility issues.', 'amp' ) }
+				</p>
+			</AdminNotice>
+		);
+	}
+
+	if ( isCompleted && pluginsWithAmpIncompatibility.length === 0 ) {
+		return (
+			<AdminNotice type={ ADMIN_NOTICE_TYPE_SUCCESS } { ...commonNoticeProps }>
+				<p>
+					{ __( 'AMP plugin found no validation errors.', 'amp' ) }
+				</p>
+			</AdminNotice>
+		);
+	}
+
+	if ( isCompleted && pluginsWithAmpIncompatibility.length > 0 ) {
+		return (
+			<AdminNotice type={ ADMIN_NOTICE_TYPE_WARNING } { ...commonNoticeProps }>
+				<p
+					dangerouslySetInnerHTML={ {
+						__html: sprintf(
+							/* translators: the placeholder is a link to the Plugin Suppression Settings panel. */
+							__( 'AMP Plugin found validation errors. <a href="%s">Review Plugin Suppression Settings</a>', 'amp' ),
+							PLUGIN_SUPPRESSION_LINK,
+						),
+					} }
+				/>
+				{ userIsTechnical && (
+					<PluginsWithAmpIncompatibility pluginsWithAmpIncompatibility={ pluginsWithAmpIncompatibility } />
+				) }
+				<div className="amp-site-scan-notice__cta">
+					<a
+						href={ AMP_COMPATIBLE_PLUGINS_URL }
+						className="button"
+						{ ...isExternalUrl( AMP_COMPATIBLE_PLUGINS_URL ) ? { target: '_blank', rel: 'noreferrer' } : {} }
+					>
+						{ __( 'View Compatible Plugins List', 'amp' ) }
+					</a>
+				</div>
+			</AdminNotice>
+		);
+	}
+
+	return (
+		<AdminNotice type={ ADMIN_NOTICE_TYPE_INFO } { ...commonNoticeProps }>
+			<p>
+				{ __( 'AMP plugin is checking your site for compatibility issues', 'amp' ) }
+				<Loading inline={ true } />
```

Comment:
Can we change the parent tag from `p` to something to `span`? because the `Loading` component uses the `div` tag which causes below error in the browser.

```
Warning: validateDOMNesting(...): <div> cannot appear as a descendant of <p>.
div
Loading@https://amp-dev.test/wp-content/plugins/amp/assets/js/amp-site-scan-notice.js?ver=f1872f78b895de49a60ee614caa3a22f:786:7
p
div
AdminNotice@https://amp-dev.test/wp-content/plugins/amp/assets/js/amp-site-scan-notice.js?ver=f1872f78b895de49a60ee614caa3a22f:367:7
SiteScanNotice@https://amp-dev.test/wp-content/plugins/amp/assets/js/amp-site-scan-notice.js?ver=f1872f78b895de49a60ee614caa3a22f:66:69
SiteScanContextProvider@https://amp-dev.test/wp-content/plugins/amp/assets/js/amp-site-scan-notice.js?ver=f1872f78b895de49a60ee614caa3a22f:1564:7
PluginsContextProvider@https://amp-dev.test/wp-content/plugins/amp/assets/js/amp-site-scan-notice.js?ver=f1872f78b895de49a60ee614caa3a22f:1154:7
UserContextProvider@https://amp-dev.test/wp-content/plugins/amp/assets/js/amp-site-scan-notice.js?ver=f1872f78b895de49a60ee614caa3a22f:1913:7
OptionsContextProvider@https://amp-dev.test/wp-content/plugins/amp/assets/js/amp-site-scan-notice.js?ver=f1872f78b895de49a60ee614caa3a22f:869:7
ErrorBoundary@https://amp-dev.test/wp-content/plugins/amp/assets/js/amp-site-scan-notice.js?ver=f1872f78b895de49a60ee614caa3a22f:530:5
ErrorContextProvider@https://amp-dev.test/wp-content/plugins/amp/assets/js/amp-site-scan-notice.js?ver=f1872f78b895de49a60ee614caa3a22f:623:7
Providers@https://amp-dev.test/wp-content/plugins/amp/assets/js/amp-site-scan-notice.js?ver=f1872f78b895de49a60ee614caa3a22f:3600:7 react-dom.860550b4.js:61:32
```

Reply:
That's a good catch!

Instead of changing the parent tag, the `<Loading inline={ true } />` component will now render as a `<span>` and not as a `<div>` (see 5253a32).